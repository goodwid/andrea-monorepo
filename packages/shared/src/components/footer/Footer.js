import React from 'react';
import PropTypes from 'prop-types';

import styles from './Footer.scss';

import {
  NavLink,
} from 'react-router-dom';

const Footer = (props) => {
  const { buttons } = props;
  return (
    <nav className={styles.Footer}>
      { buttons.map((button, index) => {
        return (
          <NavLink exact key={index} to={button.to}
            className="nav-button" 
            aria-label={button.label}
            activeClassName="active">
            <button.Icon className="icon" />
            {button.title}
          </NavLink>
        );
      })}
    </nav>
  );
};

Footer.propTypes = {
  buttons: PropTypes.array.isRequired,
};

export default Footer;
