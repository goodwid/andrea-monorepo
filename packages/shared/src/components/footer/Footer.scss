@import 'shared/styles/variables.scss';

:local(.Footer) {
  font-family: $header-font;
  display: flex;
  align-items: center;
  justify-content: space-around;
  height: $footer-size;
  width: 100vw;
  position: fixed;
  bottom: 0;
  background-color: $white;
  border-top: 1px solid $light-gray;

  a {
    display: flex;
    flex-direction: column;
    align-items: center;
    color: gray;
    text-decoration: none;
    transition: .2s ease-in-out;
  }

  .active {
    color: $sky-blue;
  }
    
  .icon {
    margin-bottom: 0.5rem;
    font-size: 2.2rem;

    @include media-for-iPhone-5SE {
      font-size: 2rem;
    }
  }
}
