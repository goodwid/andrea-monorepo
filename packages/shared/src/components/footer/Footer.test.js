import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

import {
  FaUser as User,
  FaList as JobList,
  FaIdBadge as JobBadge,
} from 'react-icons/fa';

const footerButtons = [
  {
    Icon: User,
    to: '/profile',
    title: 'Profile',
    label: 'Visit your profile to edit information about yourself',
  },
  {
    Icon: JobList,
    to: '/',
    title: 'Job List',
    label: 'Visit the job queue to see a list of available jobs',
  },
  {
    Icon: JobBadge,
    to: '/myjobs',
    title: 'My Jobs',
    label: 'Visit your jobs list to see a list of your completed, applied and scheduled jobs',
  },
];

describe('Footer page', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<Footer buttons={footerButtons} />);
    expect(wrapper).toMatchSnapshot();
  });
});
