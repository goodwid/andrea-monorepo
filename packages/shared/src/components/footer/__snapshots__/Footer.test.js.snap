// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer page matches the snapshot 1`] = `
<nav
  className="Footer"
>
  <NavLink
    activeClassName="active"
    aria-label="Visit your profile to edit information about yourself"
    className="nav-button"
    exact={true}
    key="0"
    to="/profile"
  >
    <FaUser
      className="icon"
    />
    Profile
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-label="Visit the job queue to see a list of available jobs"
    className="nav-button"
    exact={true}
    key="1"
    to="/"
  >
    <FaList
      className="icon"
    />
    Job List
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-label="Visit your jobs list to see a list of your completed, applied and scheduled jobs"
    className="nav-button"
    exact={true}
    key="2"
    to="/myjobs"
  >
    <FaIdBadge
      className="icon"
    />
    My Jobs
  </NavLink>
</nav>
`;
